### ISSUE TITLE
Bug: predict time is not stable across calls

### ISSUE THREAD

**Issue by cakedev0** [probabl-ai/skore#2671]
The cache slot for the predict time can be overwritten by calling a metric that relies on a `response_method` that was not in cache yet. For instance:

```Python
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression
from skore import evaluate

X, y = make_classification()
report = evaluate(LogisticRegression(), X, y, splitter=0.2)
report.metrics.roc_auc()
print(report.metrics.timings())
# {'fit_time': 0.001524532970506698, 'predict_time_test': 0.0005574490060098469}
report.metrics.accuracy()
print(report.metrics.timings())
# {'fit_time': 0.001524532970506698, 'predict_time_test': 0.000334909011144191}
```

I'd say that this isn't a desired behavior. 

The expected behavior needs decision, we can either:
- only record the time when `response_method="predict"`
- only record the time for the first predictions (`response_method="predict"/"predict_proba"/"decision_function"`)
- have "predict_time_{data_source}" + optionally "predict_proba_time_{data_source}" and/or  "decision_function_time_{data_source}"

I think all options are valid because prediction speed seems to be the same for the different respond methods, for what I could try. But options 1 or 3 feels a bit more rigorous.

**Comment by GaetandeCast** [comment]
To me, the issue here is that `accuracy()` calls `predict()` instead of thresholding the cached `predict_proba`. IMO we should cache the `predict_proba`/`decision_function` when available and use it everywhere; and only rely on `predict` when it's the only prediction method available.
